xbmc/filesystem/HTTPDirectory.cpp

```diff
@@ -98,8 +98,18 @@ bool CHTTPDirectory::GetDirectory(const CURL& url, CFileItemList &items)
       std::string strLinkBase = strLink;
       std::string strLinkOptions;
 
+      //convert to URL encode
+      size_t pos = strLinkBase.find_first_of("+;");
+      while (pos != std::string::npos) 
+      {
+        std::stringstream convert;
+        convert << '%' << std::hex << int(strLinkBase.at(pos));
+        strLinkBase.replace(pos, 1, convert.str());
+        pos = strLinkBase.find_first_of("+;");
+      }
```

Comment:
the whole block should be moved down after stripping off the query string (StrLinkOptions), otherwise you encode a "+", which is a reserved char(for space) in a query string only.

Reply:
Yes you're right I'll make the correction shortly

Reply:
Yes you're right I'll make the correction shortly